src/ray/object_manager/object_manager.cc

```diff
@@ -193,19 +190,30 @@ void ObjectManager::TryPull(const ObjectID &object_id) {
     return;
   }
 
+  auto &client_vector = it->second.client_locations;
+
   // The timer should never fire if there are no expected client locations.
-  RAY_CHECK(!it->second.client_locations.empty());
+  RAY_CHECK(!client_vector.empty());
   RAY_CHECK(local_objects_.count(object_id) == 0);
-
-  // Get the next client to try.
-  const ClientID client_id = std::move(it->second.client_locations.back());
-  it->second.client_locations.pop_back();
+  // Make sure that there is at least one client which is not the local client.
+  // TODO(rkn): It may actually be possible for this check to fail.
```

Comment:
What does this TODO mean? When would the check fail?

Reply:
I'm imagining a scenario where
1. The object is created locally.
2. The object manager adds it to the object table
3. The object is evicted.
4. We call `Pull` on the object, causing us to subscribe to the object table.
5. We then get the notification from step 2 about the object being created locally.

Now that I write it down, it sounds impossible, but I remember thinking in the past that the guarantees between the ordering of publishes and subscriptions were not as strong as we'd like. @pcmoritz do you remember what I'm talking about?